classes/manager.php

```diff
@@ -455,4 +455,105 @@ public static function require_auth($courseorid = null, $autologinguest = null,
             }
         }
     }
+
+    /**
+     * Sets config variable for given factor.
+     *
+     * @param array $data
+     * @param string $factor
+     *
+     * @return bool true or exception
+     * @throws dml_exception
+     */
+    public static function set_factor_config($data, $factor) {
+        foreach ($data as $key => $newvalue) {
+            $oldvalue = get_config($factor, $key);
+            if ($oldvalue != $newvalue) {
+                set_config($key, $newvalue, $factor);
+                add_to_config_log($key, $oldvalue, $newvalue, $factor);
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Checks if MFA Plugin is enabled and has enabled factor.
+     * If plugin is disabled or there is no enabled factors,
+     * it means there is nothing to do from user side.
+     * Thus, login flow shouldn't be extended with MFA.
+     *
+     * @return bool
+     * @throws \dml_exception
+     */
+    public static function mfa_ready() {
```

Comment:
Nit: the mfa prefix is probably redundant here if we were writing this from scratch, I'd call it is_ready